src/utils/hibp.ts

```diff
@@ -211,13 +212,41 @@ export type HibpLikeDbBreach = {
 /* c8 ignore start */
 async function getAllBreachesFromDb(): Promise<HibpLikeDbBreach[]> {
   let dbBreaches: BreachRow[] = [];
+  const redisBreachKey = "breaches";
+  const rClient = redisClient();
+
   try {
-    dbBreaches = await getAllBreaches();
+    // attempts to fetch breaches from Redis first
+    let redisBreaches;
+    if (rClient) {
+      redisBreaches = JSON.parse(
+        (await rClient.get(redisBreachKey)) || "[]",
+      ) as BreachRow[];
+    }
+
+    if (!redisBreaches || redisBreaches.length < 1) {
+      // if Redis fails, attempt to get breaches from Postgres and set Redis
```

Comment:
I think this will happen more often than you expect, because there's a lag between the time the cron job runs that ingests these and the time that breach alerts come in.

I'm not sure what the best way to fix this is, we don't want to have a "thundering herd" problem where this code tries to refresh the cache.

Maybe it'd be best to return `429` if Redis is down, and not do the fallback to Postgres?

Reply:
I don't think this is that big of a deal. Without Redis, all the nodes currently do GET from Postgres. Without the fallback, users will not get the right breaches information here. The top priority for us is to provide the correct information to users.

I'm adding the logic to `set` Redis in another PR via the cronjob, this way the cache should be kept fresh, and it's unlikely this code triggers. But I think it's important to have this safeguard here regardless

Reply:
See the description of the PR for more info / clarification